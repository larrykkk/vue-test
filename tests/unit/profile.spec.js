import { mount } from "@vue/test-utils";
import Profile from "@/components/profile.vue";

describe("Profile.vue", () => {
  it("當 isAdmin 為 true，將顯示: 哈囉! ${username}", () => {
    const wrapper = mount(Profile, {
      propsData: {
        isAdmin: true,
        username: "larry",
      },
    });
    expect(wrapper.text()).toMatch("哈囉! larry");
  });

  it("當 isAdmin 為 false，將顯示: 沒有權限訪問", () => {
    const wrapper = mount(Profile, {
      propsData: {
        isAdmin: false,
      },
    });
    expect(wrapper.text()).toMatch("沒有權限訪問");
  });
});
