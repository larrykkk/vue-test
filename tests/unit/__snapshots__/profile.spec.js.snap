// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue 當 isAdmin 為 false，將顯示: 沒有權限訪問 1`] = `
<div>
  <div>
    沒有權限訪問
  </div>
   
  <button>
    back
  </button>
</div>
`;
